### PR TITLE
PLAT-692: implement VCallRequest validation

### PR DESCRIPTION
**Pull request by bigbes**
subj

### CODE REVIEW

File: ledger-core/insolar/payload/validation_vcallrequest.go [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+	var (
+		callerPulse = m.Caller.GetLocal().GetPulseNumber()
+	)
+	if !isSpecialOrTimePulseBeforeOrEq(callerPulse, currentPulse) {
+		return throw.New("Caller should have special or valid time pulse before or equal to current pulse")
+	}
+
+	if !m.Callee.IsSelfScope() {
+		return throw.New("")
```
Comment (cyraxred) [inline ledger-core/insolar/payload/validation_vcallrequest.go:99]: empty error name?

File: ledger-core/insolar/payload/validation_vcallrequest.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	}
+
+	var (
+		calleePulse = m.Caller.GetLocal().GetPulseNumber()
+	)
+	if !isSpecialOrTimePulseBeforeOrEq(calleePulse, currentPulse) {
+		return throw.New("")
```
Comment (cyraxred) [inline ledger-core/insolar/payload/validation_vcallrequest.go:106]: if this is not an accidental, then it is better to use a dedicated error instance

File: ledger-core/insolar/payload/validation_vcallrequest.go (hunk 1 above)
Comment (bigbes) [reply ledger-core/insolar/payload/validation_vcallrequest.go]: accident

File: ledger-core/insolar/payload/validation_vcallrequest.go (hunk 2 above)
Comment (bigbes) [reply ledger-core/insolar/payload/validation_vcallrequest.go]: accident

**Pull request merged**